pkg/process/config/yaml_config.go

```diff
@@ -80,6 +80,16 @@ func (a *AgentConfig) loadSysProbeYamlConfig(path string) error {
 		a.ConntrackRateLimit = config.Datadog.GetInt(key(spNS, "conntrack_rate_limit"))
 	}
 
+	// When reading kernel structs at different offsets, don't go over the threshold
+	// This defaults to 400 and has a max of 3000. These are arbitrary choices to avoid infinite loops.
+	if th := config.Datadog.GetInt(key(spNS, "offset_guess_threshold")); th > 0 {
+		if th < maxOffsetThreshold {
+			a.OffsetGuessThreshold = uint64(th)
+		} else {
+			log.Warn("Overriding the configured offset_guess_threshold because it exceeds maximum of 3000")
```

Comment:
If the setting is exceeded, should you set `a.OffsetGuessThreshold` to the default(400) or to the max(3000)?
Also, I feel like the word `overriding` here is a bit misleading, I'd maybe reword and print the actual value used for the setting.

Reply:
Hey @truthbk, in other situations in which we have set a value too high, we use the default, so I was maintaining that behavior for consistency's sake.